src/target/adiv5.c

```diff
@@ -289,32 +272,62 @@ static const struct {
 };
 
 #ifdef ENABLE_DEBUG
-const char *adiv5_ap_type[] = {
-	"AHB3-AP",
-	"APB2/3-AP",
-	"AXI3/4-AP",
-	"AHB5-AP",
-	"APB4/5-AP",
-	"AXI5-AP",
-	"AHB5-AP",
-};
-
-const char *adiv5_ap_map_type(const uint8_t ap_type, const uint8_t ap_class)
+static const char *adiv5_arm_ap_type_string(const uint8_t ap_type, const uint8_t ap_class)
 {
-	/* Type 0 APs are determined by the class code */
-	if (ap_type == 0U) {
+	/*
+	 * Values taken from ADIv5 spec §C1.3 pg146.
+	 * table C1-2 "AP Identification types for an AP designed by Arm"
+	 */
+
+	switch (ap_type) {
```

Comment:
This is looking much better. Given the `ap_class` is supposed to be 8 per the table for this first block of `ap_type`'s, it is probably a good idea to pivot this slightly to first checking `ap_class` then switching on `ap_type` (and yes, this was a flaw in our original logic too), but we'd be happy merging this as-is if you don't want to go through that to make this exactingly correct.

Reply:
Good catch, added the additional validation